### ISSUE TITLE
Feature request: Detect some simple/common shell injection casese

### ISSUE THREAD

**Issue by charles-dyfis-net** [koalaman/shellcheck#1889]
#### For new checks and feature suggestions
- [X] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [X] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh
xargs -I{} sh -c 'echo {}'
```

...or...

```sh
#!/bin/sh
sh -c "echo $1"
```

#### Here's what shellcheck currently says:

```
No issues detected!
```

#### Here's what I wanted or expected to see:

Textually substituting results into the script argument of `sh` runs serious security risks. Keep your script argument constant, and add parameters after that position.

```sh
#!/bin/sh
sh -c 'echo "$1"' _ "$1"
```

```sh
xargs sh -c 'for arg do echo "$arg"; done' _
```
